torch/ao/quantization/fx/_convert_do_not_use.py

```diff
@@ -55,6 +78,138 @@
+            if len(users) > 1:
+                for user in users:
+                    with quantized.graph.inserting_before(node):
+                        new_node = quantized.graph.create_node("call_method", "dequantize", node.args, {})
+                    user.replace_input_with(node, new_node)
+                quantized.graph.erase_node(node)
+
+    quantized = QuantizedGraphModule(quantized_root, quantized.graph, quantized_root.preserved_attr_names)
+    return quantized
+
+def remove_extra_dequantize(quantized: QuantizedGraphModule) -> QuantizedGraphModule:
+    """
+    Removes duplicate dequant nodes in the graph, for an operator that has multiple dequant nodes as a user,
+    replace them with a single dequant node that can be shared across all the uses.
+    """
+    quantized_root = quantized
+    for node in quantized.graph.nodes:
+        users = list(node.users)
+        dequant_users = [user for user in node.users if user.op == "call_method" and user.target == "dequantize" or
+                         (user.op == "call_function" and user.target == torch.dequantize)]
+
+        if len(dequant_users) > 1:
+            with quantized.graph.inserting_after(node):
+                unique_dq = quantized.graph.create_node("call_method", "dequantize", users[0].args, {})
+            for dequant in dequant_users:
+                dequant.replace_all_uses_with(unique_dq)
+                quantized.graph.erase_node(dequant)
+
+    quantized = QuantizedGraphModule(quantized_root, quantized.graph, quantized_root.preserved_attr_names)
+    return quantized
+
+def remove_quant_dequant_pairs(quantized: QuantizedGraphModule) -> QuantizedGraphModule:
+    quantized_root = quantized
+    for node in quantized.graph.nodes:
+        if node.op == "call_function" and node.target in [torch.quantize_per_tensor, torch.quantize_per_channel]:
+            users = list(node.users)
+            user = users[0] if users else None
+            if len(users) == 1 and user.op == "call_method" and user.target == "dequantize":
+                user.replace_all_uses_with(node.args[0])
+                quantized.graph.erase_node(user)
+                orig_args = list(node.args)
+                quantized.graph.erase_node(node)
+                for arg in orig_args:
+                    if isinstance(arg, Node) and len(list(arg.users)) == 0:
+                        quantized.graph.erase_node(arg)
+
+    quantized = QuantizedGraphModule(quantized_root, quantized.graph, quantized_root.preserved_attr_names)
+    return quantized
+
+def get_module_path_and_prefix(
+        obs_node: Node,
+        node_name_to_scope: Dict[str, Tuple[str, type]],
+        qconfig_map: Dict[str, QConfigAny]):
+    # an observer can be inserted for both input of the next operator or output of the previous
+    # operator (they can be the same)
+    # this flag identifies if the observer is inserted only because the observed node is
+    # the input of the next operator
```

Comment:
Maybe add a docstring like the other functions? Also what is `this_flag` referring to?